lib/irq.h

```diff
@@ -19,21 +19,59 @@ extern "C" {
 /** \defgroup irq Interrupt Handling Interfaces
  *  @{ */
 
+#include <metal/list.h>
 #include <stdlib.h>
 
 /** IRQ handled status */
 #define METAL_IRQ_NOT_HANDLED 0
 #define METAL_IRQ_HANDLED     1
 
+/** IRQ ANY ID */
+#define METAL_IRQ_ANY         (-1)
+
+/** IRQ state macro which will be passed to metal irq set state function
+ *  to indicate which state the caller want the IRQ to change to.
+ */
+#define METAL_IRQ_ENABLE     1U
+#define METAL_IRQ_DISABLE    2U
+#define METAL_IRQ_PENDING    4U
+
+struct metal_device;
```

Comment:
This is not used by irq.  can we remove it?

Reply:
Will remove the metal_device